### DISCUSSION TITLE
Annke C800 Turret Camera

### DISCUSSION THREAD

**Discussion by mmmmmcoke** [blakeblackshear/frigate#1276] [Ask A Question]
Well, I ended up picking up one of the Annke C800 Turret cameras that went on sale recently.
But I can't seem to get it to play nice with Frigate.  I just get a green screen.   Also the debug logs show 0 for all of the fps counters.

Has anyone else had any luck with the Annke C800 cameras?  I'm wondering if it is due to the fact that it only supports H265/H265+.... no H264.

Config Exerpt:
    "driveway_cam": {
      "best_image_timeout": 60,
      "clips": {
        "enabled": true,
        "objects": null,
        "post_capture": 5,
        "pre_capture": 5,
        "required_zones": [],
        "retain": {
          "default": 7,
          "objects": {}
        }
      },
      "detect": {
        "enabled": true,
        "max_disappeared": 25
      },
      "ffmpeg_cmds": [
        {
          "cmd": "ffmpeg -hide_banner -loglevel warning -avoid_negative_ts make_zero -fflags +genpts+discardcorrupt -rtsp_transport tcp -stimeout 5000000 -use_wallclock_as_timestamps 1 -i rtsp://admin:word0Fblake!@192.168.108.56:554/Streaming/Channels/101 -f segment -segment_time 10 -segment_format mp4 -reset_timestamps 1 -strftime 1 -c copy -an /tmp/cache/driveway_cam-%Y%m%d%H%M%S.mp4 -c copy -f flv rtmp://127.0.0.1/live/driveway_cam -r 5 -f rawvideo -pix_fmt yuv420p pipe:",
          "roles": [
            "detect",
            "clips",
            "rtmp"
          ]
        }
      ],
      "fps": 5,
      "frame_shape": [
        2160,
        3840
      ],
      "height": 2160,
      "motion": {
        "contour_area": 100,
        "delta_alpha": 0.2,
        "frame_alpha": 0.2,
        "frame_height": 360,
        "mask": null,
        "threshold": 25
      },
      "mqtt": {
        "bounding_box": true,
        "crop": true,
        "enabled": true,
        "height": 270,
        "required_zones": [],
        "timestamp": true
      },
      "name": "driveway_cam",
      "objects": {
        "filters": {
          "bicycle": {
            "mask": [],
            "max_area": 24000000,
            "min_area": 0,
            "min_score": 0.5,
            "threshold": 0.7
          },
          "car": {
            "mask": [],
            "max_area": 24000000,
            "min_area": 0,
            "min_score": 0.5,
            "threshold": 0.7
          },
          "cat": {
            "mask": [],
            "max_area": 24000000,
            "min_area": 0,
            "min_score": 0.5,
            "threshold": 0.7
          },
          "dog": {
            "mask": [],
            "max_area": 24000000,
            "min_area": 0,
            "min_score": 0.5,
            "threshold": 0.7
          },
          "motorcycle": {
            "mask": [],
            "max_area": 24000000,
            "min_area": 0,
            "min_score": 0.5,
            "threshold": 0.7
          },
          "person": {
            "mask": [],
            "max_area": 24000000,
            "min_area": 0,
            "min_score": 0.5,
            "threshold": 0.7
          }
        },
        "mask": null,
        "track": [
          "person",
          "bicycle",
          "car",
          "motorcycle",
          "cat",
          "dog"
        ]
      },
      "record": {
        "enabled": false,
        "retain_days": 30
      },
      "rtmp": {
        "enabled": true
      },
      "snapshots": {
        "bounding_box": true,
        "crop": false,
        "enabled": true,
        "height": null,
        "required_zones": [],
        "retain": {
          "default": 7,
          "objects": {}
        },
        "timestamp": false
      },
      "width": 3840,
      "zones": {}
    },

**Comment by pr0927** [comment]
Check out my issue thread, #1178, as that's likely of interest to you for this camera and the whole H.265/H.265+ dynamic.

**Reply by mmmmmcoke** [reply]
Thanks for the help.  I hopped into your thread and tried disabling rtmp and that got the feed up.  Now I need to see how I can get hw acceleration working.  I'll follow along in the thread you started.

**Reply by pr0927** [reply]
Sure thing, happy to have (possibly) helped you solve this too! Fingers-crossed our experimenting gets everything working perfectly.

**Reply by snzsnz** [reply]
@pr0927  - Could you post up your c800 config? I've still having trouble with mine even after doing what you said above. 
No audio in recordings if I dont use any output args. If I do, then no recording at all. Can stream it live fine (HD with audio)

**Reply by pr0927** [reply]
> @pr0927 - Could you post up your c800 config? I've still having trouble with mine even after doing what you said above. No audio in recordings if I dont use any output args. If I do, then no recording at all. Can stream it live fine (HD with audio)

Sure thing, here's one:

```
  patio:
    zones:
      # Required: name of the zone
      # NOTE: This must be different than any camera names, but can match with another zone on another camera.
      patio_interior:
        # Required: List of x,y coordinates to define the polygon of the zone.
        # NOTE: Coordinates can be generated at https://www.image-map.net/
        coordinates: 885,720,878,583,856,355,827,189,662,117,538,61,263,110,45,178,0,225,0,720
      front_steps:
        # Required: List of x,y coordinates to define the polygon of the zone.
        # NOTE: Coordinates can be generated at https://www.image-map.net/
        coordinates: 1183,649,987,659,995,720,1280,720,1280,491
    # Required: ffmpeg settings for the camera
    # Optional: camera level motion config
    motion:
      # Optional: motion mask
      # NOTE: see docs for more detailed info on creating masks
      mask:
        - 805,720,810,596,800,522,671,512,520,513,492,594,467,720
        - 1280,382,942,394,883,186,530,35,201,104,36,165,0,204,0,0,1280,0
    ffmpeg:
      hwaccel_args:
        - -c:v
        - hevc_cuvid
      input_args:
        - -avoid_negative_ts
        - make_zero
        - -fflags
        - +genpts+discardcorrupt
        - -rtsp_transport
        - tcp
        - -timeout
        - '5000000'
        - -use_wallclock_as_timestamps
        - '1'
      inputs:
        - path: 
            rtsp://username:password@192.168.30.52:554/Streaming/Channels/101?transportmode=unicast&profile=Profile_1
          roles:
            - record
        - path: 
            rtsp://username:password@192.168.30.52:554/Streaming/Channels/102?transportmode=unicast&profile=Profile_1
          roles:
            - detect
      output_args:
          # Optional: output args for record streams (default: shown below)
        record: -f segment -segment_time 60 -segment_format mp4 -reset_timestamps
          1 -strftime 1 -c:v h264_nvenc -crf 18 -c:a copy -vf format=nv12 -b:v 4M
          -r 15
          # Optional: output args for rtmp streams (default: shown below)
          #rtmp: -c:v h264_vaapi -c:a copy -vf format=nv12|vaapi,hwupload -f flv
    best_image_timeout: 15
    record:
      enabled: true
      retain:
        # Optional: Number of days to retain recordings regardless of events (default: shown below)
        # NOTE: This should be set to 0 and retention should be defined in events section below
        #       if you only want to retain recordings of events.
        days: 10
        # Optional: Mode for retention. Available options are: all, motion, and active_objects
        #   all - save all recording segments regardless of activity
        #   motion - save all recordings segments with any detected motion
        #   active_objects - save all recording segments with active/moving objects
        # NOTE: this mode only applies when the days setting above is greater than 0
        mode: all
      alerts:
        retain: {}
      detections:
        retain: {}
    detect:
      width: 1280
      height: 720
      fps: 5
    objects:
      track:
        - person
        - cat
        - dog
        - bird
        - bicycle
    snapshots:
      # Optional: Enable writing jpg snapshot to /media/frigate/clips (default: shown below)
      # This value can be set via MQTT and will be updated in startup based on retained value
      enabled: true
      # Optional: print a timestamp on the snapshots (default: shown below)
      timestamp: false
      # Optional: draw bounding box on the snapshots (default: shown below)
      bounding_box: true
      # Optional: crop the snapshot (default: shown below)
      crop: false
```
